packages/builders/src/interactions/commands/chatInput/Assertions.ts

```diff
@@ -0,0 +1,139 @@
+import {
+	ApplicationIntegrationType,
+	InteractionContextType,
+	ApplicationCommandOptionType,
+} from 'discord-api-types/v10';
+import type { ZodTypeAny } from 'zod';
+import { z } from 'zod';
+import { localeMapPredicate, memberPermissionsPredicate } from '../../../Assertions.js';
+import { applicationCommandOptionAllowedChannelTypes } from './mixins/ApplicationCommandOptionChannelTypesMixin.js';
+
+export const namePredicate = z
+	.string()
+	.min(1)
+	.max(32)
+	.regex(/^[\p{Ll}\p{Lm}\p{Lo}\p{N}\p{sc=Devanagari}\p{sc=Thai}_-]+$/u);
+
+export const descriptionPredicate = z.string().min(1).max(100);
+
+export const sharedNameAndDescriptionPredicate = z.object({
+	name: namePredicate,
+	name_localizations: localeMapPredicate.optional(),
+	description: descriptionPredicate,
+	description_localizations: localeMapPredicate.optional(),
+});
+
+export const numericMixinNumberOptionPredicate = z.object({
+	max_value: z.number().optional(),
+	min_value: z.number().optional(),
+});
+
+export const numericMixinIntegerOptionPredicate = z.object({
+	max_value: z.number().int().optional(),
+	min_value: z.number().int().optional(),
+});
+
+export const channelMixinOptionPredicate = z.object({
+	channel_types: z
+		.union(
+			applicationCommandOptionAllowedChannelTypes.map((type) => z.literal(type)) as unknown as [
+				ZodTypeAny,
+				ZodTypeAny,
+				...ZodTypeAny[],
+			],
+		)
+		.array()
+		.optional(),
+});
+
+export const autocompleteMixinOptionPredicate = z.object({
+	autocomplete: z.literal(true),
+	choices: z.union([z.never(), z.never().array(), z.undefined()]),
+});
+
+export const choiceStringPredicate = z.string().min(1).max(100);
+export const choiceNumberPredicate = z.number().min(Number.NEGATIVE_INFINITY).max(Number.POSITIVE_INFINITY);
```

Comment:
https://github.com/discord/discord-api-spec/blob/348aeb6011459285685c9a4db7989a6fd4cbf508/specs/openapi.json#L10242-L10251

```suggestion
export const choiceNumberPredicate = z.number().safe();
```